crates/compiler/mono/src/borrow.rs

```diff
@@ -136,37 +150,25 @@ impl<'a> DeclarationToIndex<'a> {
         needle_symbol: Symbol,
         needle_layout: ProcLayout<'a>,
     ) -> ParamOffset {
-        if let Ok(middle_index) = self
-            .elements
-            .binary_search_by_key(&needle_symbol, |t| t.0 .0)
-        {
-            // first, iterate backward until we hit a different symbol
-            let backward = self.elements[..middle_index].iter().rev();
-
-            for ((symbol, proc_layout), param_offset) in backward {
-                if *symbol != needle_symbol {
-                    break;
-                } else if *proc_layout == needle_layout {
-                    return *param_offset;
-                }
-            }
-
-            // if not found, iterate forward until we find our combo
-            let forward = self.elements[middle_index..].iter();
-
-            for ((symbol, proc_layout), param_offset) in forward {
-                if *symbol != needle_symbol {
-                    break;
-                } else if *proc_layout == needle_layout {
-                    return *param_offset;
-                }
-            }
+        if let Some(param_offset) = self.elements.get(&Declaration {
+            symbol: needle_symbol,
+            layout: needle_layout,
+        }) {
+            return *param_offset;
         }
 
         let similar = self
             .elements
             .iter()
-            .filter_map(|((s, lay), _)| if *s == needle_symbol { Some(lay) } else { None })
+            .filter_map(
+                |(
+                    Declaration {
+                        symbol: s,
+                        layout: lay,
+                    },
+                    _,
+                )| if *s == needle_symbol { Some(lay) } else { None },
```

Comment:
Since I use a `Declaration` to enable a custom `Hash` implementation I don't use a tuple. But removing the existing aliases convinced the linter to put it on a single line.